src/applications/hca/containers/IDPage.jsx

```diff
@@ -104,15 +190,26 @@ const mapDispatchToProps = {
   toggleLoginModal,
 };
 
-const mapStateToProps = state => ({
-  enrollmentStatus: state.hcaIDForm.enrollmentStatus,
-  errors: state.hcaIDForm.errors,
-  form: state.form,
-  noESRRecordFound: state.hcaIDForm.noESRRecordFound,
-  isSubmittingIDForm: state.hcaIDForm.isSubmitting,
-  shouldRedirect: isLoggedIn(state),
-  showLoadingIndicator: isProfileLoading(state),
-});
+const mapStateToProps = state => {
+  const {
+    enrollmentStatus,
+    hasServerError,
+    isSubmitting,
+    loginRequired,
+    noESRRecordFound,
+  } = state.hcaIDForm;
+  return {
+    enrollmentStatus,
+    form: state.form,
+    isSubmittingIDForm: isSubmitting,
+    loginRequired,
+    noESRRecordFound,
+    shouldRedirect: isLoggedIn(state),
+    showContinueButton: !loginRequired,
+    showLoadingIndicator: isProfileLoading(state),
+    showServerError: hasServerError,
```

Comment:
+1

Reply:
Looks like there might be some `ownProps` passed into this that could be documented in a `propTypes` object. I've also seen `ownProps` plucked out in `mapStateToProps` so props from state and own props are assigned in the same place- not sure how that works exactly.

Reply:
Will add `propTypes`. Do you have any thoughts on documenting the props that come from `mapStateToProps` in `propTypes` as well? It looks like something we haven't really done much in the past, but I know it's doable.

Reply:
Yes. The propTypes list should include every prop. 

The way I see it, mapStateToProps is part of the react-redux connect component- not the component it wraps. It’s a helper that takes the state and transforms it to the propsTypwa of the component and handles componentShouldUpdate for you automatically.